### ISSUE TITLE
Firefox throws TypeError for invalid URLs

### ISSUE THREAD

**Issue by filoozom** [pouchdb/pouchdb#4595]
This issue only occurs on Firefox, and works as expected in all the other browsers I tried.
When you try to replicate to an invalid URL, you get an error on https://github.com/pouchdb/pouchdb/blob/d2ec5f9e3ecff4f373e88473260208e586ddd998/lib/replicate/replicate.js#L219 : 
```
TypeError: setting a property that has only a getter
```

```console.log(err);``` gives something like : 
```
Exception { message: "", result: 2152398858, name: "NS_ERROR_MALFORMED_URI", filename: "http://cdn.jsdelivr.net/pouchdb/lat…", lineNumber: 9, columnNumber: 0, inner: null, data: null, stack: "i@http://cdn.jsdelivr.net/pouchdb/l…" }
```

You can replicate the error on Firefox by providing an invalid URL, like :
```
PouchDB('test').replicate.to('http:///')
```

**Comment by nolanlawson** [comment]
This is a good first-timers bug, because it should be very very easy to reproduce since you kindly provided the one-liner test case:

```js
new PouchDB('test').replicate.to('http:///')
```

If anybody wants to take this up, they are more than welcome. Presumably you would just need to add a test to one of our integration tests (I recommend `test.basics.js`), then follow the instructions in TESTING.md to run the tests, and then fix the issue. It should throw an error, but the error should be the same in all browsers and should be something useful to users (e.g. `"Your URL is invalid"`).

**Comment by daedlock** [comment]
@nolanlawson, I am working on closing this issue. I added url validation in `replicate.to` and `replicate.from` that throws an error on invalid urls. The problem is some tests fail because they are using invalid urls for replication and yet the error gets thrown. Please advise

**Comment by nolanlawson** [comment]
fixed by d3aa2c92b0a5bf461384184211bde68ae4037eec
